id.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2014 - Max Ekman <max@looplab.se>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eventhorizon
+
+import (
+	"fmt"
+
+	"gopkg.in/mgo.v2/bson"
+)
+
+// ID is an unique identifier type
+type ID struct {
```

Comment:
This is really tricky! I have played with your ID interface test branch for a while now, without much success of finding a solution to it. I do get why a wrapping struct ID type could be the best way forward, but I would really like to avoid the dependencies on bson and other DB specific packages for the core types. Not sure how to achieve both!